dockerfile_scripts/install_deepspeed.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEBIAN_FRONTEND=noninteractive apt-get install -y pdsh libaio-dev
+if [ "$DEEPSPEED_PIP" ]; then
+    python -m  pip install ninja psutil packaging attrdict
```

Comment:
`ninja`, `psutil`, and `packaging` seem to be deepspeed dependencies. so maybe explicitly `pip installing` them is not necessary because they'll be pulled in by deepspeed?
not sure about `attrdict`. why do you need to explicitly install it?

Reply:
i removed the deepspeed dependencies since they'll install when installing deepspeed.  i moved `attrdict` to our `additional_requirements` as a quality of life improvement when working with hyperparameters (we can access them as attributes instead of indexing with dict keys).